javascript/pull-view.js

```diff
@@ -165,14 +168,51 @@
       } else {
         text = 'Unknown';
       }
-
       // if status is success but PR is not mergable, overwrite status...
       if (success && info.get('mergeable') === false){
         classes = 'not-mergeable';
         text = 'Merge Conflicts';
       }
-
       return '<span class="status ' + classes + '">' + text + '</span>';
+    },
+
+    generateApprovalHTML: function(info,reviews) {
+      var classes = '';
+      var text = '';
+
+      var reviewers = reviews.get('reviewers')
+
+      if (reviewers != null){
+          var totalReviews = Object.keys(reviewers).length;
+          var approved = 0;
+          var requestedChanges = 0;
+          var commented = 0;
+
+          for(var review in reviewers) {
+            if(reviewers[review].state == "APPROVED"){
+                approved ++;
+            }
+            else if (reviewers[review].state == "CHANGES_REQUESTED"){
+                requestedChanges ++;
+            }
+            else if (reviewers[review].state == "COMMENTED"){
+                commented ++;
+            }
+          }
+
+          text = 'Approvals (' + approved + '/' + totalReviews + ')';
+      } else {
+        text = 'Unknown';
+      }
+
+      // if status is approved only if the no other statuses are present
+      if (approved >= 2 && requestedChanges == 0 && commented == 0 ){
+        classes = 'approved';
+      }
+      else if (approved < 2 || requestedChanges > 0 || commented > 0 ){
```

Comment:
if requestedChanges > 0 then it should be red. Consider putting that statement first e.g.

```
if (requestedChanges > 0) {
    classes = 'changes-requested';
}
else if (approved < 2) {
    classes = 'pending-approval';
}
else {
    classes = 'approved';
}
```